src/main/java/com/icthh/xm/tmf/ms/communication/rules/businesstime/BusinessTimeRule.java

```diff
@@ -33,19 +36,57 @@ public RuleResponse validate(CommunicationMessage message) {
-            BusinessTime businessTime = businessDayConfig.getCurrentBusinessTime(currentDateTime);
 
-            LocalTime currentTime = currentDateTime.toLocalTime();
-            LocalTime startTime = businessTime.getStartTime();
-            LocalTime endTime = businessTime.getEndTime();
-            log.debug("start time: {}, end time: {}, current time: {}", startTime, endTime, currentTime);
+            boolean exceptionCharacteristicPresent = false;
+            if (businessDayConfig.getBusinessTime() != null) {
+                exceptionCharacteristicPresent = checkMessageCharacteristics(message, businessDayConfig);
+            }
+
+            if (!exceptionCharacteristicPresent){
+                BusinessTime businessTime = businessDayConfig.getCurrentBusinessTime(currentDateTime);
+
+                LocalTime currentTime = currentDateTime.toLocalTime();
+                LocalTime startTime = businessTime.getStartTime();
+                LocalTime endTime = businessTime.getEndTime();
+                log.debug("start time: {}, end time: {}, current time: {}", startTime, endTime, currentTime);
 
-            if (startTime.isAfter(currentTime) || endTime.isBefore(currentTime)) {
-                ruleResponse.setSuccess(false);
-                ruleResponse.setResponseCode(NOT_BUSINESS_TIME_CODE);
-                return ruleResponse;
+                if (startTime.isAfter(currentTime) || endTime.isBefore(currentTime)) {
+                    ruleResponse.setSuccess(false);
+                    ruleResponse.setResponseCode(NOT_BUSINESS_TIME_CODE);
+                    return ruleResponse;
+                }
             }
         }
         return ruleResponse;
     }
+
+    /**
+     * Check whether the message contains characteristics which are allow to send it at not business time
+     * @param message Message to check
+     * @param businessDayConfig Configuration
+     * @return Whether the message contains characteristics which are allow to send it at not business time
+     */
+    private boolean checkMessageCharacteristics(CommunicationMessage message, BusinessDayConfig businessDayConfig) {
+        boolean exceptionCharacteristicPresent = false;
+        List<CommunicationRequestCharacteristic> exceptionCharacteristics = businessDayConfig.getBusinessTime()
+            .getExceptionCharacteristics();
+
+        if (exceptionCharacteristics != null && !exceptionCharacteristics.isEmpty() && message.getCharacteristic() != null) {
+            exceptionCharacteristicPresent = message.getCharacteristic().stream()
+                .anyMatch(exceptionCharacteristics::contains);
+
+            if (exceptionCharacteristicPresent && log.isDebugEnabled()) {
+                // Map message to one string for Kibana and like a Json array for pretty look
+                log.debug(message.getCharacteristic().stream().filter(exceptionCharacteristics::contains)
+                    .map(
```

Comment:
Do you really need a JSON as a format of characteristics log output? Json concatenation logic seems too complex. 
Could you consider to collect `characteristic` to Map<String, Object> (key = characteristic.getName(), value = characteristic.getValue()) and just print this map?

Reply:
In the message these objects are the List, so what a purpose to display it in the log as map?  The developers will become confused  at the log analyzing.

Reply:
@AnPogrebnyak i was confused by constructing JSON with string concatenation. 
as a result output will be like 
`[{"name"=x, "value"=y}, {...}]`
With proposed Map solution output would be like 
`{x=y, ...}` which is almost the same but printing code is much simpler.